scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -286,17 +286,20 @@ object TreeSyntax {
               s("(())")
             case (arg: Term) :: Nil =>
               val needsParens = arg match {
-                // Include parentheses for `a op (apply(b, _))`
-                case apply: Term.Apply if apply.args.exists {
+                case apply: Term.Apply if apply.args.exists { // `a op (apply(b, _))`
                     case _: Term.Placeholder => true
                     case _ => false
-                  } =>
+                  } => true
+                case Term.Select(_: Term.Placeholder, _) => // `a op (_.b)`
                   true
-                // Include parentheses for `a op (bar: _*)`
-                case repeated: Term.Repeated =>
+                case Term.Apply(Term.Select(lhs: Term.Placeholder, _), _) => // `a op (_.b(c))`
                   true
-                case _ =>
+                case Term.ApplyInfix(lhs, _, _, _) if !lhs.isInstanceOf[Term.Placeholder] => // Not `a op (_ b c)`
+                  false
+                case _: Lit | _: Term.Ref | _: Term.Function | _: Term.Apply | _: Term.If | _: Term.Match =>
```

Comment:
Is `Term.Match` correct here?
```scala
scala> List(1) map _ match { case 1 => println("BOOM") }
<console>:12: error: missing parameter type for expanded function
The argument types of an anonymous function must be fully known. (SLS 8.5)
Expected type was: ?
       List(1) map _ match { case 1 => println("BOOM") }
                   ^

scala> List(1) map (_ match { case 1 => println("BOOM") })
BOOM
res1: List[Unit] = List(())
```

Reply:
I am not entirely sure, but it seems to want to add double parentheses without it. I've added a test for that case though (and there were some others)